Rendered fewer hooks than expected. This may be caused by an accidental early return statement.

Related: https://github.com/toeverything/AFFiNE/pull/2257

![image](https://user-images.githubusercontent.com/14026360/236737887-bfa0bd35-d0be-49fd-b453-ecdf7564f90a.png)

```
index.ts:22 Debug logs enabled
eval @ index.ts:22
reactive-element.js:45 Lit is in dev mode. Not recommended for production! See https://lit.dev/msg/dev-mode for more information.
issueWarning @ reactive-element.js:45
config.ts:149 Project: @affine/web
config.ts:150 Build date: 5/7/2023, 11:58:43 PM
config.ts:154 Editor Version: 0.0.0-20230508043859-34d0cc68-nightly
config.ts:156 Version: v0.0.1-alpha-Abbey-Wood-1608-gac96da8a
config.ts:157 AFFiNE is an open source project, you can view its source code on GitHub!
config.ts:160 https://github.com/toeverything/AFFiNE/tree/ac96da8a
_app.tsx:1712 Runtime Preset Object
_app.tsx:1712 Runtime Preset Object
common.js:113 local-provider [INFO] connect indexeddb provider +0ms NyfjUoPxcP
:8080/chrome-192x192.png:1     Failed to load resource: the server responded with a status of 404 (Not Found)
common.js:113 web:atoms:root [INFO] workspaces +0ms Array(1)
common.js:113 local-provider [INFO] connect broadcast channel +187ms NyfjUoPxcP
common.js:113 local-provider [INFO] connect indexeddb provider +1ms NyfjUoPxcP
common.js:113 workspace-layout [INFO] mount +0ms
common.js:113 local-provider [INFO] disconnect broadcast channel +1ms NyfjUoPxcP
common.js:113 local-provider [INFO] disconnect indexeddb provider +0ms NyfjUoPxcP
common.js:113 workspace-layout [INFO] unmount +1ms
common.js:113 local-provider [INFO] connect broadcast channel +1ms NyfjUoPxcP
common.js:113 local-provider [INFO] connect indexeddb provider +1ms NyfjUoPxcP
common.js:113 workspace-layout [INFO] mount +2ms
client.js:1 workspace-layout [ERROR] list data error: +0ms Error: Not implemented
    at unimplemented (index.tsx:107:11)
    at fetch (workspace-layout.tsx:9652:77)
console.error @ client.js:1
client.js:1 workspace-layout [ERROR] list data error: +1ms Error: Not implemented
    at unimplemented (index.tsx:107:11)
    at fetch (workspace-layout.tsx:9652:77)
console.error @ client.js:1
common.js:113 workspace-layout [INFO] mount first data: +0ms Array(1)
common.js:113 local-provider [INFO] connect indexeddb provider +1ms NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-page-id.ts:17 cleanup page
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-page-id.ts:17 cleanup page
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
common.js:113 local-provider [INFO] connect indexeddb provider +52ms NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
use-sync-router-with-current-workspace-id.ts:42 set workspace id NyfjUoPxcP
common.js:113 web:atoms:root [INFO] workspaces +286ms Array(1)
common.js:113 local-provider [INFO] disconnect broadcast channel +635ms NyfjUoPxcP
common.js:113 local-provider [INFO] disconnect indexeddb provider +0ms NyfjUoPxcP
common.js:113 workspace-layout [INFO] currentWorkspace:  +798ms Object
common.js:113 local-provider [INFO] connect broadcast channel +111ms NyfjUoPxcP
common.js:113 local-provider [INFO] connect indexeddb provider +1ms NyfjUoPxcP
common.js:113 local-provider [INFO] disconnect broadcast channel +1ms NyfjUoPxcP
common.js:113 local-provider [INFO] disconnect indexeddb provider +0ms NyfjUoPxcP
common.js:113 workspace-layout [INFO] currentWorkspace:  +2ms Object
common.js:113 local-provider [INFO] connect broadcast channel +0ms NyfjUoPxcP
common.js:113 local-provider [INFO] connect indexeddb provider +1ms NyfjUoPxcP
client.js:1 Using kebab-case for css properties in objects is not supported. Did you mean WebkitUserSelect?
console.error @ client.js:1
client.js:1 Using kebab-case for css properties in objects is not supported. Did you mean WebkitUserDrag?
console.error @ client.js:1
client.js:1 Using kebab-case for css properties in objects is not supported. Did you mean WebkitAppRegion?
console.error @ client.js:1
client.js:1 Warning: React has detected a change in the order of Hooks called by ResizeIndicator. This will lead to bugs and errors if not fixed. For more information, read the Rules of Hooks: https://reactjs.org/link/rules-of-hooks

   Previous render            Next render
   ------------------------------------------------------
1. useMemo                    useMemo
2. useContext                 useContext
3. useCallback                useCallback
4. useContext                 useContext
5. useReducer                 useReducer
6. useEffect                  useEffect
7. useDebugValue              useDebugValue
8. useContext                 useContext
9. useCallback                useCallback
10. useState                  useState
11. useEffect                 useEffect
12. useContext                useContext
13. useReducer                useReducer
14. useEffect                 useEffect
15. useDebugValue             useDebugValue
16. useContext                useDeferredValue
   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    at ResizeIndicator (webpack-internal:///../../packages/component/src/components/app-sidebar/resize-indicator/index.tsx:1369:107)
    at RootAppSidebar (webpack-internal:///./src/components/root-app-sidebar/index.tsx:5446:11)
    at div
    at eval (webpack-internal:///../../node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js:60:66)
    at WorkspaceLayoutInner (webpack-internal:///./src/layouts/workspace-layout.tsx:9840:11)
    at Provider (webpack-internal:///./src/plugins/local/index.tsx:548:19)
    at Suspense
    at CurrentWorkspaceContext (webpack-internal:///./src/layouts/workspace-layout.tsx:9493:11)
    at WorkspaceLayout (webpack-internal:///./src/layouts/workspace-layout.tsx:9587:11)
    at f (webpack-internal:///../../node_modules/next-themes/dist/index.module.js:8:597)
    at $ (webpack-internal:///../../node_modules/next-themes/dist/index.module.js:8:348)
    at ThemeProvider (webpack-internal:///./src/providers/ThemeProvider.tsx:822:11)
    at DebugProvider (webpack-internal:///./src/pages/_app.tsx:1640:11)
    at Provider (webpack-internal:///../../node_modules/jotai/esm/react.mjs:20:3)
    at ProviderComposer (webpack-internal:///./src/components/provider-composer.tsx:305:11)
    at Suspense
    at AffineErrorBoundary (webpack-internal:///./src/components/affine/affine-error-eoundary.tsx:742:9)
    at I18nextProvider (webpack-internal:///../../node_modules/react-i18next/dist/es/I18nextProvider.js:10:19)
    at App (webpack-internal:///./src/pages/_app.tsx:1678:11)
    at PathnameContextProviderAdapter (webpack-internal:///../../node_modules/next/dist/shared/lib/router/adapters.js:1446:11)
    at ErrorBoundary (webpack-internal:///../../node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:303:63)
    at ReactDevOverlay (webpack-internal:///../../node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:852:919)
    at Container (webpack-internal:///../../node_modules/next/dist/client/index.js:8174:1)
    at AppContainer (webpack-internal:///../../node_modules/next/dist/client/index.js:8380:11)
    at Root (webpack-internal:///../../node_modules/next/dist/client/index.js:8683:11)
console.error @ client.js:1
client.js:1 Warning: The final argument passed to useEffect changed size between renders. The order and size of this array must remain constant.

Previous: [[object Object], atom29, ]
Incoming: [256,[object Object],atom29]
    at ResizeIndicator (webpack-internal:///../../packages/component/src/components/app-sidebar/resize-indicator/index.tsx:1369:107)
    at RootAppSidebar (webpack-internal:///./src/components/root-app-sidebar/index.tsx:5446:11)
    at div
    at eval (webpack-internal:///../../node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js:60:66)
    at WorkspaceLayoutInner (webpack-internal:///./src/layouts/workspace-layout.tsx:9840:11)
    at Provider (webpack-internal:///./src/plugins/local/index.tsx:548:19)
    at Suspense
    at CurrentWorkspaceContext (webpack-internal:///./src/layouts/workspace-layout.tsx:9493:11)
    at WorkspaceLayout (webpack-internal:///./src/layouts/workspace-layout.tsx:9587:11)
    at f (webpack-internal:///../../node_modules/next-themes/dist/index.module.js:8:597)
    at $ (webpack-internal:///../../node_modules/next-themes/dist/index.module.js:8:348)
    at ThemeProvider (webpack-internal:///./src/providers/ThemeProvider.tsx:822:11)
    at DebugProvider (webpack-internal:///./src/pages/_app.tsx:1640:11)
    at Provider (webpack-internal:///../../node_modules/jotai/esm/react.mjs:20:3)
    at ProviderComposer (webpack-internal:///./src/components/provider-composer.tsx:305:11)
    at Suspense
    at AffineErrorBoundary (webpack-internal:///./src/components/affine/affine-error-eoundary.tsx:742:9)
    at I18nextProvider (webpack-internal:///../../node_modules/react-i18next/dist/es/I18nextProvider.js:10:19)
    at App (webpack-internal:///./src/pages/_app.tsx:1678:11)
    at PathnameContextProviderAdapter (webpack-internal:///../../node_modules/next/dist/shared/lib/router/adapters.js:1446:11)
    at ErrorBoundary (webpack-internal:///../../node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:303:63)
    at ReactDevOverlay (webpack-internal:///../../node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:852:919)
    at Container (webpack-internal:///../../node_modules/next/dist/client/index.js:8174:1)
    at AppContainer (webpack-internal:///../../node_modules/next/dist/client/index.js:8380:11)
    at Root (webpack-internal:///../../node_modules/next/dist/client/index.js:8683:11)
console.error @ client.js:1
client.js:1 Warning: The final argument passed to useEffect changed size between renders. The order and size of this array must remain constant.

Previous: [[object Object], atom29, ]
Incoming: [256,[object Object],atom29]
    at ResizeIndicator (webpack-internal:///../../packages/component/src/components/app-sidebar/resize-indicator/index.tsx:1369:107)
    at RootAppSidebar (webpack-internal:///./src/components/root-app-sidebar/index.tsx:5446:11)
    at div
    at eval (webpack-internal:///../../node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js:60:66)
    at WorkspaceLayoutInner (webpack-internal:///./src/layouts/workspace-layout.tsx:9840:11)
    at Provider (webpack-internal:///./src/plugins/local/index.tsx:548:19)
    at Suspense
    at CurrentWorkspaceContext (webpack-internal:///./src/layouts/workspace-layout.tsx:9493:11)
    at WorkspaceLayout (webpack-internal:///./src/layouts/workspace-layout.tsx:9587:11)
    at f (webpack-internal:///../../node_modules/next-themes/dist/index.module.js:8:597)
    at $ (webpack-internal:///../../node_modules/next-themes/dist/index.module.js:8:348)
    at ThemeProvider (webpack-internal:///./src/providers/ThemeProvider.tsx:822:11)
    at DebugProvider (webpack-internal:///./src/pages/_app.tsx:1640:11)
    at Provider (webpack-internal:///../../node_modules/jotai/esm/react.mjs:20:3)
    at ProviderComposer (webpack-internal:///./src/components/provider-composer.tsx:305:11)
    at Suspense
    at AffineErrorBoundary (webpack-internal:///./src/components/affine/affine-error-eoundary.tsx:742:9)
    at I18nextProvider (webpack-internal:///../../node_modules/react-i18next/dist/es/I18nextProvider.js:10:19)
    at App (webpack-internal:///./src/pages/_app.tsx:1678:11)
    at PathnameContextProviderAdapter (webpack-internal:///../../node_modules/next/dist/shared/lib/router/adapters.js:1446:11)
    at ErrorBoundary (webpack-internal:///../../node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:303:63)
    at ReactDevOverlay (webpack-internal:///../../node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:852:919)
    at Container (webpack-internal:///../../node_modules/next/dist/client/index.js:8174:1)
    at AppContainer (webpack-internal:///../../node_modules/next/dist/client/index.js:8380:11)
    at Root (webpack-internal:///../../node_modules/next/dist/client/index.js:8683:11)
console.error @ client.js:1
react-dom.development.js:16381 Uncaught Error: Rendered fewer hooks than expected. This may be caused by an accidental early return statement.
    at renderWithHooks (react-dom.development.js:16381:1)
    at updateFunctionComponent (react-dom.development.js:19588:1)
    at beginWork (react-dom.development.js:21601:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:4164:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:1)
    at invokeGuardedCallback (react-dom.development.js:4277:1)
    at beginWork$1 (react-dom.development.js:27451:1)
    at performUnitOfWork (react-dom.development.js:26557:1)
    at workLoopSync (react-dom.development.js:26466:1)
    at renderRootSync (react-dom.development.js:26434:1)
    at performConcurrentWorkOnRoot (react-dom.development.js:25738:1)
    at workLoop (scheduler.development.js:266:1)
    at flushWork (scheduler.development.js:239:1)
    at MessagePort.performWorkUntilDeadline (scheduler.development.js:533:1)
client.js:1 Warning: The final argument passed to useEffect changed size between renders. The order and size of this array must remain constant.

Previous: [[object Object], atom29, ]
Incoming: [256,[object Object],atom29]
    at ResizeIndicator (webpack-internal:///../../packages/component/src/components/app-sidebar/resize-indicator/index.tsx:1369:107)
    at RootAppSidebar (webpack-internal:///./src/components/root-app-sidebar/index.tsx:5446:11)
    at div
    at eval (webpack-internal:///../../node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js:60:66)
    at WorkspaceLayoutInner (webpack-internal:///./src/layouts/workspace-layout.tsx:9840:11)
    at Provider (webpack-internal:///./src/plugins/local/index.tsx:548:19)
    at Suspense
    at CurrentWorkspaceContext (webpack-internal:///./src/layouts/workspace-layout.tsx:9493:11)
    at WorkspaceLayout (webpack-internal:///./src/layouts/workspace-layout.tsx:9587:11)
    at f (webpack-internal:///../../node_modules/next-themes/dist/index.module.js:8:597)
    at $ (webpack-internal:///../../node_modules/next-themes/dist/index.module.js:8:348)
    at ThemeProvider (webpack-internal:///./src/providers/ThemeProvider.tsx:822:11)
    at DebugProvider (webpack-internal:///./src/pages/_app.tsx:1640:11)
    at Provider (webpack-internal:///../../node_modules/jotai/esm/react.mjs:20:3)
    at ProviderComposer (webpack-internal:///./src/components/provider-composer.tsx:305:11)
    at Suspense
    at AffineErrorBoundary (webpack-internal:///./src/components/affine/affine-error-eoundary.tsx:742:9)
    at I18nextProvider (webpack-internal:///../../node_modules/react-i18next/dist/es/I18nextProvider.js:10:19)
    at App (webpack-internal:///./src/pages/_app.tsx:1678:11)
    at PathnameContextProviderAdapter (webpack-internal:///../../node_modules/next/dist/shared/lib/router/adapters.js:1446:11)
    at ErrorBoundary (webpack-internal:///../../node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:303:63)
    at ReactDevOverlay (webpack-internal:///../../node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:852:919)
    at Container (webpack-internal:///../../node_modules/next/dist/client/index.js:8174:1)
    at AppContainer (webpack-internal:///../../node_modules/next/dist/client/index.js:8380:11)
    at Root (webpack-internal:///../../node_modules/next/dist/client/index.js:8683:11)
console.error @ client.js:1
client.js:1 Warning: The final argument passed to useEffect changed size between renders. The order and size of this array must remain constant.

Previous: [[object Object], atom29, ]
Incoming: [256,[object Object],atom29]
    at ResizeIndicator (webpack-internal:///../../packages/component/src/components/app-sidebar/resize-indicator/index.tsx:1369:107)
    at RootAppSidebar (webpack-internal:///./src/components/root-app-sidebar/index.tsx:5446:11)
    at div
    at eval (webpack-internal:///../../node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js:60:66)
    at WorkspaceLayoutInner (webpack-internal:///./src/layouts/workspace-layout.tsx:9840:11)
    at Provider (webpack-internal:///./src/plugins/local/index.tsx:548:19)
    at Suspense
    at CurrentWorkspaceContext (webpack-internal:///./src/layouts/workspace-layout.tsx:9493:11)
    at WorkspaceLayout (webpack-internal:///./src/layouts/workspace-layout.tsx:9587:11)
    at f (webpack-internal:///../../node_modules/next-themes/dist/index.module.js:8:597)
    at $ (webpack-internal:///../../node_modules/next-themes/dist/index.module.js:8:348)
    at ThemeProvider (webpack-internal:///./src/providers/ThemeProvider.tsx:822:11)
    at DebugProvider (webpack-internal:///./src/pages/_app.tsx:1640:11)
    at Provider (webpack-internal:///../../node_modules/jotai/esm/react.mjs:20:3)
    at ProviderComposer (webpack-internal:///./src/components/provider-composer.tsx:305:11)
    at Suspense
    at AffineErrorBoundary (webpack-internal:///./src/components/affine/affine-error-eoundary.tsx:742:9)
    at I18nextProvider (webpack-internal:///../../node_modules/react-i18next/dist/es/I18nextProvider.js:10:19)
    at App (webpack-internal:///./src/pages/_app.tsx:1678:11)
    at PathnameContextProviderAdapter (webpack-internal:///../../node_modules/next/dist/shared/lib/router/adapters.js:1446:11)
    at ErrorBoundary (webpack-internal:///../../node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:303:63)
    at ReactDevOverlay (webpack-internal:///../../node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:852:919)
    at Container (webpack-internal:///../../node_modules/next/dist/client/index.js:8174:1)
    at AppContainer (webpack-internal:///../../node_modules/next/dist/client/index.js:8380:11)
    at Root (webpack-internal:///../../node_modules/next/dist/client/index.js:8683:11)
console.error @ client.js:1
react-dom.development.js:16381 Uncaught Error: Rendered fewer hooks than expected. This may be caused by an accidental early return statement.
    at renderWithHooks (react-dom.development.js:16381:1)
    at updateFunctionComponent (react-dom.development.js:19588:1)
    at beginWork (react-dom.development.js:21601:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:4164:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:1)
    at invokeGuardedCallback (react-dom.development.js:4277:1)
    at beginWork$1 (react-dom.development.js:27451:1)
    at performUnitOfWork (react-dom.development.js:26557:1)
    at workLoopSync (react-dom.development.js:26466:1)
    at renderRootSync (react-dom.development.js:26434:1)
    at recoverFromConcurrentError (react-dom.development.js:25850:1)
    at performConcurrentWorkOnRoot (react-dom.development.js:25750:1)
    at workLoop (scheduler.development.js:266:1)
    at flushWork (scheduler.development.js:239:1)
    at MessagePort.performWorkUntilDeadline (scheduler.development.js:533:1)
client.js:1 The above error occurred in the <ResizeIndicator> component:

    at ResizeIndicator (webpack-internal:///../../packages/component/src/components/app-sidebar/resize-indicator/index.tsx:1369:107)
    at RootAppSidebar (webpack-internal:///./src/components/root-app-sidebar/index.tsx:5446:11)
    at div
    at eval (webpack-internal:///../../node_modules/@emotion/react/dist/emotion-element-c39617d8.browser.esm.js:60:66)
    at WorkspaceLayoutInner (webpack-internal:///./src/layouts/workspace-layout.tsx:9840:11)
    at Provider (webpack-internal:///./src/plugins/local/index.tsx:548:19)
    at Suspense
    at CurrentWorkspaceContext (webpack-internal:///./src/layouts/workspace-layout.tsx:9493:11)
    at WorkspaceLayout (webpack-internal:///./src/layouts/workspace-layout.tsx:9587:11)
    at f (webpack-internal:///../../node_modules/next-themes/dist/index.module.js:8:597)
    at $ (webpack-internal:///../../node_modules/next-themes/dist/index.module.js:8:348)
    at ThemeProvider (webpack-internal:///./src/providers/ThemeProvider.tsx:822:11)
    at DebugProvider (webpack-internal:///./src/pages/_app.tsx:1640:11)
    at Provider (webpack-internal:///../../node_modules/jotai/esm/react.mjs:20:3)
    at ProviderComposer (webpack-internal:///./src/components/provider-composer.tsx:305:11)
    at Suspense
    at AffineErrorBoundary (webpack-internal:///./src/components/affine/affine-error-eoundary.tsx:742:9)
    at I18nextProvider (webpack-internal:///../../node_modules/react-i18next/dist/es/I18nextProvider.js:10:19)
    at App (webpack-internal:///./src/pages/_app.tsx:1678:11)
    at PathnameContextProviderAdapter (webpack-internal:///../../node_modules/next/dist/shared/lib/router/adapters.js:1446:11)
    at ErrorBoundary (webpack-internal:///../../node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:303:63)
    at ReactDevOverlay (webpack-internal:///../../node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:852:919)
    at Container (webpack-internal:///../../node_modules/next/dist/client/index.js:8174:1)
    at AppContainer (webpack-internal:///../../node_modules/next/dist/client/index.js:8380:11)
    at Root (webpack-internal:///../../node_modules/next/dist/client/index.js:8683:11)

React will try to recreate this component tree from scratch using the error boundary you provided, AffineErrorBoundary.
console.error @ client.js:1
client.js:1 Uncaught error: Error: Rendered fewer hooks than expected. This may be caused by an accidental early return statement.
    at renderWithHooks (react-dom.development.js:16381:1)
    at updateFunctionComponent (react-dom.development.js:19588:1)
    at beginWork (react-dom.development.js:21601:1)
    at beginWork$1 (react-dom.development.js:27426:1)
    at performUnitOfWork (react-dom.development.js:26557:1)
    at workLoopSync (react-dom.development.js:26466:1)
    at renderRootSync (react-dom.development.js:26434:1)
    at recoverFromConcurrentError (react-dom.development.js:25850:1)
    at performConcurrentWorkOnRoot (react-dom.development.js:25750:1)
    at workLoop (scheduler.development.js:266:1)
    at flushWork (scheduler.development.js:239:1)
    at MessagePort.performWorkUntilDeadline (scheduler.development.js:533:1) Object
console.error @ client.js:1
common.js:113 workspace-layout [INFO] unmount +39ms
common.js:113 local-provider [INFO] disconnect broadcast channel +38ms NyfjUoPxcP
common.js:113 local-provider [INFO] disconnect indexeddb provider +0ms NyfjUoPxcP
DevTools failed to load source map: Could not load content for chrome-extension://hnfanknocfeofbddgcijnmhnfnkdnaad/requestProvider.js.map: System error: net::ERR_BLOCKED_BY_CLIENT
DevTools failed to load source map: Could not load content for chrome-extension://hnfanknocfeofbddgcijnmhnfnkdnaad/requestSolanaProvider.js.map: System error: net::ERR_BLOCKED_BY_CLIENT
DevTools failed to load source map: Could not load content for chrome-extension://hnfanknocfeofbddgcijnmhnfnkdnaad/content.js.map: System error: net::ERR_BLOCKED_BY_CLIENT
```

https://github.com/toeverything/AFFiNE/blob/31cccafb40ca8be0f922af029466e0e1c603f7d8/packages/component/src/components/app-sidebar/resize-indicator/index.tsx

--- a/packages/forgetti/src/core/checks.ts
+++ b/packages/forgetti/src/core/checks.ts
@@ -72,3 +72,22 @@ export function isNestedExpression(node: t.Node): node is NestedExpression {
       return false;
   }
 }
+
+/** Check whether a Node is CallExpression and a React Hook call */
+export function isCallExpressionAndHook(ctx: StateContext, node: t.Node): node is t.CallExpression {
+  if (node.type !== 'CallExpression') {
+    return false;
+  }
+
+  // A simple check to see if "node.callee" is an Identifier
+  // TypeScript is able to infer the type of "node.callee" to Identifier or V8IntrinsicIdentifier
+  if (!('name' in node.callee)) {
+    return false;
+  }
+
+  // Check if callee is a react hook
+  if (!ctx.filters.hook) {
+    return false;
+  }
+  return ctx.filters.hook.test(node.callee.name);
+}

--- a/packages/forgetti/src/core/optimizer.ts
+++ b/packages/forgetti/src/core/optimizer.ts
@@ -1,7 +1,7 @@
 /* eslint-disable @typescript-eslint/no-unnecessary-condition */
 import type * as babel from '@babel/core';
 import * as t from '@babel/types';
-import { isNestedExpression, isPathValid } from './checks';
+import { isCallExpressionAndHook, isNestedExpression, isPathValid } from './checks';
 import getForeignBindings, { isForeignBinding } from './get-foreign-bindings';
 import getImportIdentifier from './get-import-identifier';
 import { RUNTIME_EQUALS } from './imports';
@@ -561,6 +561,10 @@ export default class Optimizer {
         for (let i = 0, len = argumentsPath.length; i < len; i++) {
           argument = argumentsPath[i];
           if (isPathValid(argument, t.isExpression)) {
+            if (isCallExpressionAndHook(this.ctx, argument.node)) {
+              continue;
+            }
+
             const optimized = this.createDependency(argument);
             if (optimized) {
               mergeDependencies(dependencies, optimized.deps);

--- a/packages/forgetti/test/hooks.test.ts
+++ b/packages/forgetti/test/hooks.test.ts
@@ -114,4 +114,15 @@ function Example(props) {
 `;
     expect(await compile(code)).toMatchSnapshot();
   });
+  it('should correct transform nested hooks call (issue #14)', async () => {
+    const code = `
+import { useDeferredValue } from 'react';
+import { useAtomValue } from 'jotai';
+import { stateAtom } from 'whatever';
+function Example(props) {
+  return useDeferredValue(useAtomValue(stateAtom));
+}
+`;
+    expect(await compile(code)).toMatchSnapshot();
+  });
 });

